File: backend/src/main/java/com/bwevent/domain/model/Banquet.java
package com.bwevent.domain.model;

import com.bwevent.domain.enums.BanquetSource;
import com.bwevent.domain.enums.BanquetStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "banquets")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Banquet {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "restaurant_id", nullable = false)
    private UUID restaurantId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "contact_id")
    private Contact contact;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private BanquetStatus status = BanquetStatus.DRAFT;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private BanquetSource source = BanquetSource.MANUAL;

    @Column(name = "event_date")
    private LocalDate date;

    @Column(name = "start_time")
    private LocalTime startTime;

    @Column(name = "end_time")
    private LocalTime endTime;

    private Integer headcount;

    @Column(precision = 10, scale = 2)
    private BigDecimal budget;

    @Column(name = "room_setup")
    private String roomSetup;

    @Column(name = "dietary_restrictions", columnDefinition = "TEXT")
    private String dietaryRestrictions;

    @Column(name = "av_needs", columnDefinition = "TEXT")
    private String avNeeds;

    @Column(name = "deposit_paid")
    @Builder.Default
    private Boolean depositPaid = false;

    @Column(name = "deposit_amount", precision = 10, scale = 2)
    private BigDecimal depositAmount;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(name = "event_type_id")
    private UUID eventTypeId;

    @Column(name = "created_by")
    private UUID createdBy;

    @OneToMany(mappedBy = "banquet", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @Builder.Default
    private List<MenuItem> menuItems = new ArrayList<>();

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Instant updatedAt;
}

File: frontend/src/types/banquet.ts
import type { Contact } from './contact'
import type { EventType } from './eventType'

export type BanquetStatus = 'DRAFT' | 'CONFIRMED' | 'CANCELLED'
export type BanquetSource = 'MANUAL' | 'EMAIL'
export type BanquetLocation = 'FLOOR' | 'BAR' | 'COUNTER'

export const BANQUET_LOCATIONS: BanquetLocation[] = ['FLOOR', 'BAR', 'COUNTER']

export const BANQUET_LOCATION_LABELS: Record<BanquetLocation, string> = {
  FLOOR: 'Étage',
  BAR: 'Bar',
  COUNTER: 'Comptoir',
}

export interface MenuItem {
  id: string
  dishName: string
  quantity: number
  notes: string | null
}

export interface MenuItemRequest {
  dishName: string
  quantity: number
  notes?: string
}

export interface BanquetSummary {
  id: string
  status: BanquetStatus
  source: BanquetSource
  date: string | null
  startTime: string | null
  endTime: string | null
  location: BanquetLocation | null
  headcount: number | null
  contactName: string | null
  contactOrganization: string | null
}

export interface Banquet {
  id: string
  restaurantId: string
  contact: Contact | null
  status: BanquetStatus
  source: BanquetSource
  date: string | null
  startTime: string | null
  endTime: string | null
  location: BanquetLocation | null
  headcount: number | null
  budget: number | null
  roomSetup: string | null
  dietaryRestrictions: string | null
  avNeeds: string | null
  depositPaid: boolean
  depositAmount: number | null
  notes: string | null
  createdBy: string | null
  menuItems: MenuItem[]
  eventType: EventType | null
  /** fieldId → value */
  fieldValues: Record<string, string>
  createdAt: string
  updatedAt: string
}

export interface BanquetRequest {
  contactId?: string
  status?: BanquetStatus
  source?: BanquetSource
  date?: string
  startTime?: string
  endTime?: string
  location?: BanquetLocation
  headcount?: number
  budget?: number
  roomSetup?: string
  dietaryRestrictions?: string
  avNeeds?: string
  depositPaid?: boolean
  depositAmount?: number
  notes?: string
  menuItems?: MenuItemRequest[]
  eventTypeId?: string
  /** fieldId → value */
  fieldValues?: Record<string, string>
}
